beacon-chain/rpc/validator_server.go

```diff
@@ -226,3 +218,23 @@ func (vs *ValidatorServer) retrieveActiveValidator(beaconState *pbp2p.BeaconStat
 	}
 	return beaconState.ValidatorRegistry[validatorIdx], nil
 }
+
+func (vs *ValidatorServer) retrieveActiveValidatorsPublicKeys(pubkeys [][]byte) ([][]byte, error) {
+	m, err := vs.beaconDB.ValidatorIndices(pubkeys)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve validators indexes: %v", err)
+	}
+	beaconState, err := vs.beaconDB.State(vs.ctx)
```

Comment:
Please do not use the service context. Require `ctx` as an argument and pass that to `State`

Reply:
done